cmdstanpy/model.py

```diff
@@ -898,9 +906,28 @@ def sample(
                 method_args=sampler_args,
                 refresh=refresh,
             )
-            runset = RunSet(
-                args=args, chains=chains, chain_ids=chain_ids, time_fmt=time_fmt
-            )  # bookkeeping object for command, result, filepaths
+
+            # 2.28 and up default: if exe compiled w/ threads, run single proc
+            # CmdStanArgs ctor validates model exe file
+            num_threads = threads_per_chain
+            one_process_per_chain = True
+            if (
+                force_one_process_per_chain is None
+                or force_one_process_per_chain is False
+            ):
+                try:
+                    info = StringIO()
+                    do_command(cmd=[str(self.exe_file), 'info'], fd_out=info)
+                    lines = info.getvalue().split('\n')
+                    if lines[3] == 'STAN_THREADS=true':
```

Comment:
I preferred the former logic which wasn't so brittle. Assuming this will be the 4th line of the output is bound to cause problems in the future

Reply:
added util method `model_info` which parses console outputs into dict of key value pairs.